source/extensions/filters/http/cache/cache_filter.h

```diff
@@ -93,12 +93,12 @@ class CacheFilter : public Http::PassThroughFilter,
   void getTrailers();
 
   // Callbacks for HttpCache to call when headers/body/trailers are ready.
-  void onHeaders(LookupResult&& result, Http::RequestHeaderMap& request_headers);
-  void onBody(Buffer::InstancePtr&& body);
+  void onHeaders(LookupResult&& result, Http::RequestHeaderMap& request_headers, bool end_stream);
+  void onBody(Buffer::InstancePtr&& bod, bool end_stream);
   void onTrailers(Http::ResponseTrailerMapPtr&& trailers);
 
   // Set required state in the CacheFilter for handling a cache hit.
-  void handleCacheHit();
+  void handleCacheHit(bool end_stream_after_headers);
```

Comment:
Why are the two `end_stream_after_headers` parameters not named just `end_stream`? They both amount to "send what you already have (headers), then end the stream", right?

Reply:
It felt awfully ambiguous here, because this function isn't clearly anything to do with headers and doesn't have headers in its arguments. It doesn't want to end_stream immediately, and it doesn't want to wait until after everything to end_stream, so... `end_stream_after_headers`.

The cost to be less ambiguous is smaller than the risk of confusion.